src/transformers/pipelines/object_detection.py

```diff
@@ -0,0 +1,178 @@
+
+
+@add_end_docstrings(PIPELINE_INIT_ARGS)
+class ObjectDetectionPipeline(Pipeline):
+    """
+    Object detection pipeline using any :obj:`AutoModelForObjectDetection`. This pipeline predicts bounding boxes of
+    objects and their classes.
+
+    This object detection pipeline can currently be loaded from :func:`~transformers.pipeline` using the following task
+    identifier: :obj:`"object-detection"`.
+
+    See the list of available models on `huggingface.co/models
+    <https://huggingface.co/models?filter=object-detection>`__.
+    """
+
+    def __init__(
+        self,
+        model: "PreTrainedModel",
+        feature_extractor: PreTrainedFeatureExtractor,
+        framework: Optional[str] = None,
+        **kwargs
+    ):
+        super().__init__(model, feature_extractor=feature_extractor, framework=framework, **kwargs)
+
+        if self.framework == "tf":
+            raise ValueError(f"The {self.__class__} is only available in PyTorch.")
+
+        requires_backends(self, "timm")
+        requires_backends(self, "vision")
+
+        self.check_model_type(MODEL_FOR_OBJECT_DETECTION_MAPPING)
+
+        self.feature_extractor = feature_extractor
+
+    @staticmethod
+    def load_image(image: Union[str, "Image.Image"]):
+        if isinstance(image, str):
+            if image.startswith("http://") or image.startswith("https://"):
+                # We need to actually check for a real protocol, otherwise it's impossible to use a local file
+                # like http_huggingface_co.png
+                image = Image.open(requests.get(image, stream=True).raw)
+            elif os.path.isfile(image):
+                image = Image.open(image)
+            else:
+                raise ValueError(
+                    f"Incorrect path or url, URLs must start with `http://` or `https://`, and {image} is not a valid path"
+                )
+        elif isinstance(image, Image.Image):
+            image = image
+        else:
+            raise ValueError(
+                "Incorrect format used for image. Should be an url linking to an image, a local path, or a PIL image."
+            )
+        image = image.convert("RGB")
+        return image
+
+    def __call__(
+        self,
+        images: Union[str, List[str], "Image", List["Image"]],
+        threshold: Optional[float] = 0.9,
+    ) -> Union[Predictions, List[Prediction]]:
+        """
+        Detect objects (bounding boxes & classes) in the image(s) passed as inputs.
+
+        Args:
+            images (:obj:`str`, :obj:`List[str]`, :obj:`PIL.Image` or :obj:`List[PIL.Image]`):
+                The pipeline handles three types of images:
+
+                - A string containing a http link pointing to an image
+                - A string containing a local path to an image
+                - An image loaded in PIL directly
+
+                The pipeline accepts either a single image or a batch of images, which must then be passed as a string.
```

Comment:
The images shouldn't necessarily be passed as a string, so I would leave that away.

Reply:
Great catch!